cli/arguments/reference.go

```diff
@@ -69,20 +73,41 @@ func ParseReference(arg string, parseArch bool) (*Reference, error) {
 		ret.Version = toks[1]
 	}
 
-	if parseArch {
-		toks = strings.Split(ret.PackageName, ":")
-		if len(toks) != 2 {
-			return nil, fmt.Errorf(tr("invalid item %s"), arg)
-		}
-		if toks[0] == "" {
-			return nil, fmt.Errorf(tr("invalid empty core name '%s'"), arg)
+	toks = strings.Split(ret.PackageName, ":")
+	if len(toks) != 2 {
+		return nil, fmt.Errorf(tr("invalid item %s"), arg)
+	}
+	if toks[0] == "" {
+		return nil, fmt.Errorf(tr("invalid empty core name '%s'"), arg)
+	}
+	ret.PackageName = toks[0]
+	if toks[1] == "" {
+		return nil, fmt.Errorf(tr("invalid empty core architecture '%s'"), arg)
+	}
+	ret.Architecture = toks[1]
+
+	// Now that we have the required informations in `ret` we can
+	// try to use core.GetPlatforms to optimize what the user typed
+	// (by replacing the PackageName and Architecture in ret with the content of core.GetPlatform())
+	platforms, _ := core.GetPlatforms(&rpc.PlatformListRequest{
+		Instance:      instance.CreateAndInit(),
+		UpdatableOnly: false,
+		All:           true, // this is true because we want also the installable platforms
+	})
+	var found = 0
+	for _, platform := range platforms {
+		if strings.EqualFold(platform.GetId(), ret.PackageName+":"+ret.Architecture) {
+			logrus.Infof("Found possible match for reference %s -> %s", arg, platform.GetId())
+			toks = strings.Split(platform.GetId(), ":")
+			found++
 		}
+	}
+	// replace the returned Reference only if only one occurrence is found, otherwise leave it as is
+	if found == 1 {
 		ret.PackageName = toks[0]
-		if toks[1] == "" {
-			return nil, fmt.Errorf(tr("invalid empty core architecture '%s'"), arg)
-		}
 		ret.Architecture = toks[1]
+	} else {
+		logrus.Warnf("Found %d platform for reference: %s", found, arg)
```

Comment:
I don't think we should just warn in this case, if we're not which platform the user actually meant to use we should return an error in my opinion.